### PR TITLE
refactor: charge gas for initialized ticks 

### PR DESCRIPTION
**Pull request by stackman27**
Closes: #[4850](https://github.com/osmosis-labs/osmosis/issues/4850) & https://github.com/osmosis-labs/osmosis/issues/4947

## What is the purpose of the change
Add gas cost while initializing ticks

## Brief Changelog
Add gas cost of 10_000 while initializing ticks

## Testing and Verifying
- tested withe minGasConsumed

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/tick_test.go
Comment (stackman27) [inline x/concentrated-liquidity/tick_test.go:99]: set as testing param, just so if we ever want to change it from constant gas cost to linearly scaling gas cost, we can just change this value in test

File: x/concentrated-liquidity/tick.go
Comment (p0mvn) [inline x/concentrated-liquidity/tick.go:63]: Let's define `10000` as a `const` at the top of the file please

File: x/concentrated-liquidity/tick_test.go
Comment (p0mvn) [inline x/concentrated-liquidity/tick_test.go:113]: Building up on my previous comment, consider importing this from `tick.go`. That way, if we end up changing that value later, we only need to update one constant and zero tests

**Review by p0mvn — APPROVED** [review]
LGTM after comments are addressed. Nice work!

**Comment by czarcas7ic** [comment]
It's likely running out of gas from making the position, which is worrisome

**Comment by stackman27** [comment]
> It's likely running out of gas from making the position, which is worrisome

added tracking issue: https://github.com/osmosis-labs/osmosis/issues/4947

**Comment by p0mvn** [comment]
> > It's likely running out of gas from making the position, which is worrisome
> 
> added tracking issue: #4947

Is it possible to address this in this PR, please? Personally, I'm not feeling great about merging something that we don't fully understand, might only cause unnecessary friction with concurrent PRs if this ends up being problematic or hard to debug

**Comment by stackman27** [comment]
> > > It's likely running out of gas from making the position, which is worrisome
> > 
> > 
> > added tracking issue: #4947
> 
> Is it possible to address this in this PR, please? Personally, I'm not feeling great about merging something that we don't fully understand, might only cause unnecessary friction with concurrent PRs if this ends up being problematic or hard to debug

sounds good! resolved. I hardcoded gas and fees

File: tests/e2e/configurer/chain/commands.go
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:89]: Why are these values chosen specifically?

Is it possible to break them down to make it explicit where `600000` gas and `5000uosmo` are coming from, please?

Comment (czarcas7ic) [reply tests/e2e/configurer/chain/commands.go:89]: If I'm not mistaken, this should only require the .0025uosmo min fee since we are using less than 1mil gas

Comment (stackman27) [reply tests/e2e/configurer/chain/commands.go:89]: these are the values closest to the ones i found on local osmosis setup.sh

Comment (stackman27) [reply tests/e2e/configurer/chain/commands.go]: @czarcas7ic so i looked it up and the fees i set to `Fees = txfeestypes.ConsensusMinFee.Mul(sdk.NewDec(GasLimit)).Ceil().TruncateInt64()` and in this case wouldnt' it be 60000 * 0.0025 = 1500uosmo

Comment (stackman27) [reply tests/e2e/configurer/chain/commands.go]: okay so redid this a bit. 
New param; 
gas = 50,000 (e2e was default to 40,000, i added extra 10k because that's the hard coded gas to consume per tick initialization) 
fees = 1250 (because 50,000 * 0.0025 = 1250) 

added in comments too

Comment (czarcas7ic) [reply tests/e2e/configurer/chain/commands.go]: Nice, this logic follows to me

**Review by czarcas7ic — APPROVED** [review]
I think we are good to merge this. The calculation that Sishir made in the comment feels correct to me, and at the very least we are not just setting this to auto and forgetting about it

File: tests/e2e/configurer/chain/commands.go
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:89]: Shouldn't we be initializing two ticks per position?

Comment (stackman27) [reply tests/e2e/configurer/chain/commands.go:89]: i confirmed in `UpdatePosition` unit test to make sure we are using 20k gas while initializing test here's the output 
```
with consume gas 
=== RUN   TestKeeperTestSuite/TestUpdatePosition/update_existing_position_with_positive_amount
BEFORE  INITIALIZE TICK (Lower & Upper) :  345134
AFTER  INITIALIZE TICK (Lower & Upper) GAS:  396742

without consume gas 
=== RUN   TestKeeperTestSuite/TestUpdatePosition/update_existing_position_with_positive_amount
BEFORE  INITIALIZE TICK (Lower & Upper) GAS:  345134
AFTER INITIALIZE TICK (Lower & Upper) GAS:  376742
```

maybe in e2e the fees were too low to begin with?

Comment (p0mvn) [reply tests/e2e/configurer/chain/commands.go:89]: Discussed offline to not block on this but keep this open question in mind. Thanks for looking into it @stackman27

**Review by p0mvn — APPROVED** [review]

**Pull request merged**
